src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -83,34 +83,56 @@ public IEnumerable<string> SupportedVersions
         /// <inheritdoc/>
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
-            return _detector.Detect(context);
+            PlatformDetectorResult detectionResult;
+            if (TryGetExplicitVersion(out var explicitVersion))
+            {
+                detectionResult = new PlatformDetectorResult
+                {
+                    Platform = DotNetCoreConstants.PlatformName,
+                    PlatformVersion = explicitVersion,
+                };
+            }
+            else
+            {
+                detectionResult = _detector.Detect(context);
+            }
```

Comment:
From this logic, we check the explicit version from here so that means we'll remove `ResolvedPlatformRuntimeVersion` from detectorContext. I'll accomodate this change as well, so we don't want to keep the DetectorContext in the library?

Reply:
The `DetectorContext ` inherits `RepositoryContext ` and I moved `RepositoryContext `into Common proj, so both library and build script generator are pointing to the same `RepositoryContext `. Because when we use it from build script generator, we want to pass in the `RepositoryContext ` parameter to detectorlibrary directly, otherwise we need to do conversion. This is the current way to handle this part.